### PR TITLE
WIP Add flattenConcat and flattenMerge.

### PR DESCRIPTION
**Pull request by He-Pin**
Only Scala api

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 1]
```diff
@@ -2173,7 +2173,9 @@ trait FlowOps[+Out, +Mat] {
    *
    * '''Cancels when''' downstream cancels
    */
-  def flatMapConcat[T, M](f: Out => Graph[SourceShape[T], M]): Repr[T] = map(f).via(new FlattenMerge[T, M](1))
+  def flatMapConcat[T, M](f: Out => Graph[SourceShape[T], M]): Repr[T] = flatMapMerge(1, f)
+
+  def flattenConcat[T, M](implicit ev: Out <:< Graph[SourceShape[T], M]): Repr[T] = flatMapConcat(ev)
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: for flatMapConcat

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 2]
```diff
@@ -2191,6 +2193,8 @@ trait FlowOps[+Out, +Mat] {
   def flatMapMerge[T, M](breadth: Int, f: Out => Graph[SourceShape[T], M]): Repr[T] =
     map(f).via(new FlattenMerge[T, M](breadth))
 
+  def flattenMerge[T, M](breadth: Int)(implicit ev: Out <:< Graph[SourceShape[T], M]): Repr[T] = flatMapMerge(breadth, ev)
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: for flatMapMerge

**Comment by raboof** [comment]
Refs #25024 - back then we landed on not adding these as there was no obvious way to provide their Java counterparts, and it wasn't so obvious the additional API surface was worth the extra convenience. Do you see a good way to translate this to Java?

**Comment by hepin1989** [comment]
It reminds me this https://github.com/akka/akka/pull/27350
Let's see what we can do

**Comment by hepin1989** [comment]
How about throw a `UnsupportedOperationException` if the type is not matched and provide `flatten*` in javadsl? @raboof

File: project/StreamOperatorsIndexGenerator.scala [hunk 3]
```diff
@@ -261,7 +263,7 @@ object StreamOperatorsIndexGenerator extends AutoPlugin {
       (description, categoryName)
     } catch {
       case NonFatal(ex) =>
-        throw new RuntimeException(s"Unable to extract details from $file", ex)
+        throw new RuntimeException(s"Unable to extract details from $file, error:${ex.getMessage}", ex)
```
Comment (giftig) [inline project/StreamOperatorsIndexGenerator.scala]: Missing a space after colon, but why add the message to the wrapper message at all? It's provided as the cause already.

Comment (hepin1989) [reply project/StreamOperatorsIndexGenerator.scala]: It's not showing up in the console.

**Comment by hepin1989** [comment]
As there is no clear  way to add the javadsl.close this.

**Pull request closed without merging**